src/platform/api/extension.d.ts

```diff
@@ -259,9 +259,11 @@ export interface IExportedKernelService {
      */
     getKernelSpecifications(refresh?: boolean): Promise<KernelConnectionMetadata[]>;
     /**
-     * Gets a list of all active kernel connections associated with a resource.
+     * Gets a list of all active kernel connections.
+     * If `uri` is undefined, then the kernel is not associated with any resource. I.e its currently not associated with any notebook in Jupyter extension.
+     * If `uri` is undefined, then the kernel is associated with the resource identified by the Uri.
      */
-    getActiveKernels(): Promise<{ metadata: KernelConnectionMetadata; uri: Uri }[]>;
+    getActiveKernels(): Promise<{ metadata: KernelConnectionMetadata; uri: Uri | undefined }[]>;
```

Comment:
When we added the `Uri` into this as part of a previous PR we lost the ability to manage existing jupyter sessions. This brings it back.